### PR TITLE
remove metric from topchef add getter setter in chef base

### PR DESCRIPTION
**Pull request by jdaly13**
realized I left out metric token out of base chef - these changes call a setter within constructor only by owner and a getter as well to send transfers

### CODE REVIEW

File: src/contracts/TopChef.sol [hunk 1]
```diff
@@ -146,7 +144,8 @@ contract TopChef is Chef {
         require(group.claimable != 0, "No claimable rewards to withdraw");
         // TODO do we want a backup in case a group looses access to their wallet
         require(group.groupAddress == _msgSender(), "Sender does not represent group");
-        _metric.transfer(group.groupAddress, group.claimable);
+        MetricToken metric = getMetricToken();
```
Comment (jdaly13) [inline src/contracts/TopChef.sol:147]: Should we be using SafeERC20 package to use SafeERC20.safeTransfer

**Comment by jdaly13** [comment]
do we want ability to destroy contract? @gnomadic @addyxvii

**Comment by gnomadic** [comment]
I don't think so right? I can't imagine a use case

File: src/contracts/Chef.sol [hunk 2]
```diff
@@ -54,6 +58,10 @@ abstract contract Chef is Ownable {
         return _rewardsActive;
     }
 
+    function getMetricToken() internal view virtual returns (MetricToken) {
```
Comment (jdaly13) [inline src/contracts/Chef.sol:61]: not sure if making this internal is necessary but thought only the contracts derived from this should be calling it

File: src/contracts/Chef.sol [hunk 3]
```diff
@@ -66,4 +74,8 @@ abstract contract Chef is Ownable {
     event Harvest(address harvester, uint256 agIndex, uint256 amount);
     event Withdraw(address withdrawer, uint256 agIndex, uint256 amount);
 
+    receive() external payable virtual {}
+
+    function withdrawMoney() public onlyOwner() {}
+
```
Comment (jdaly13) [inline src/contracts/Chef.sol:80]: these two have blank implementations since an abstract contract cannot receive funds but those who inherit from can

**Review by addyxvii — APPROVED** [review]

**Pull request merged**